### PR TITLE
smoke: Add functionality to collect statedumps on hangs

### PR DESCRIPTION
**Pull request by pranithk**
Signed-off-by: Pranith Kumar K pkarampu@redhat.com

### CODE REVIEW

File: smoke.sh [hunk 1]
```diff
@@ -59,6 +59,10 @@ function watchdog ()
 
     echo "Kicking in watchdog after $1 secs";
 
+    local mount_pid=$(ps auxww | grep glusterfs | grep -E "volfile-id[ =]/?$V " | awk '{print $2}' | head -1)
```
Comment (nigelbabu) [inline smoke.sh]: What's going on here? Why are you killing the mounts?

File: smoke.sh [hunk 2]
```diff
@@ -68,6 +72,8 @@ function finish ()
     if [ $RET -ne 0 ]; then
         cat /build/dbench-logs || true
     fi
+    #Move statedumps to be archived
+    mv /var/run/gluster/*dump* /var/log/glusterfs/
```
Comment (nigelbabu) [inline smoke.sh]: Does it make more sense to set the mount path to /var/log/glusterfs instead?

File: smoke.sh (hunk 1 above)
Comment (pranithk) [inline smoke.sh]: we give a signal SIGUSR1 on which gluster saves the statedump to a file.

File: smoke.sh (hunk 2 above)
Comment (pranithk) [inline smoke.sh]: We just want the statedumps to be collected as part of the tarball. Not sure I understand the intention of making it a mount. Could you explain more?

File: smoke.sh (hunk 1 above)
Comment (nigelbabu) [inline smoke.sh]: Why not just do a `gluster volume statedump` which is the documented way to get a statedump?

File: smoke.sh (hunk 2 above)
Comment (nigelbabu) [inline smoke.sh]: What I mean is to do this at the start of the test:
```
gluster volume set test-volume server.statedump-path /var/log/glusterfs
```
This avoids having to perform an mv at all.

Comment (pranithk) [inline smoke.sh]: That doesn't work for mounts. It works in conjunction with 'gluster volume statedump' which is for bricks

**Pull request merged**